src/state/slices/addressBookSlice/addressBookSlice.ts

```diff
@@ -0,0 +1,62 @@
+import type { PayloadAction } from '@reduxjs/toolkit'
+import { createSlice } from '@reduxjs/toolkit'
+import type { ChainId } from '@shapeshiftoss/caip'
+
+export type AddressBookEntry = {
+  id: string
+  name: string
+  address: string
+  chainId: ChainId
+  createdAt: number
+}
+
+export type AddressBookState = {
+  byId: Record<string, AddressBookEntry>
+  ids: string[]
+}
+
+export const initialState: AddressBookState = {
+  byId: {},
+  ids: [],
+}
+
+export const addressBookSlice = createSlice({
+  name: 'addressBook',
+  initialState,
+  reducers: create => ({
+    addAddress: create.reducer(
+      (state, action: PayloadAction<Omit<AddressBookEntry, 'id' | 'createdAt'>>) => {
+        // Extra safety just in case logic is broken on UI side and the user adds the same address twice
+        const isDuplicate = state.ids.some(existingId => {
+          const existing = state.byId[existingId]
+          return (
+            existing.chainId === action.payload.chainId &&
+            existing.address === action.payload.address
+          )
+        })
+        if (isDuplicate) return
+
+        const now = Date.now()
+        const id = `${action.payload.chainId}_${action.payload.address}_${now}`
+        const entry: AddressBookEntry = {
+          ...action.payload,
+          id,
+          createdAt: now,
+        }
+        state.byId[id] = entry
+        state.ids.push(id)
+      },
+    ),
+    deleteAddress: create.reducer((state, action: PayloadAction<string>) => {
+      delete state.byId[action.payload]
+      state.ids = state.ids.filter(id => id !== action.payload)
+    }),
+    clear: create.reducer(() => initialState),
+  }),
+  selectors: {
+    selectAddressBookEntries: state => state.ids.map(id => state.byId[id]),
```

Comment:
recommendation: to keep things declarative and readable (I am gnomes reading this without third coffee yet, which is a good Litmus test of approachability here) could use `Address` vernacular across the board or something like that.
Two hard things in SW ikik, but just feels like `Entries` may be too lower level and not convey enough?

Or perhaps I'm overthinking this and you can reply "gnomes silence pls robot"